sycl/include/CL/sycl/detail/sampler_impl.hpp

```diff
@@ -25,7 +25,7 @@ class sampler_impl {
 public:
 #ifdef __SYCL_DEVICE_ONLY__
   __ocl_sampler_t m_Sampler;
-  sampler_impl(__ocl_sampler_t Sampler) : m_Sampler(Sampler) {}
+  sampler_impl(const __ocl_sampler_t &Sampler) : m_Sampler(Sampler) {}
```

Comment:
Do you know why we have this warning? `m_Sampler` is not reference :)

Reply:
This looks like a workaround to silence the warning, rather than a proper fix.